test/model/src/core/kv_table.cpp

```diff
@@ -227,6 +227,59 @@ kv_table::remove(kv_transaction_ptr txn, const data_value &key)
     }
 }
 
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
+ */
+int
+kv_table::truncate(const data_value &start, const data_value &stop, timestamp_t timestamp)
+{
+    std::lock_guard lock_guard(_lock);
+    if (start != model::NONE && stop != model::NONE && start > stop)
+        throw model_exception("The start and the stop key are not in the right order");
+
+    auto start_iter = start == model::NONE ? _data.begin() : _data.lower_bound(start);
+    auto stop_iter = stop == model::NONE ? _data.end() : _data.upper_bound(stop);
+
+    try {
+        for (auto i = start_iter; i != stop_iter; i++)
+            i->second.add_update(
+              std::move(kv_update(NONE, fix_timestamp(timestamp))), false, false);
+    } catch (wiredtiger_exception &e) {
+        return e.error();
+    }
+
+    return 0;
+}
+
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
```

Comment:
Should we mention we are taking a transaction here? Just to differentiate with the other `kv_table::truncate`?

Reply:
I updated the comments.

That being said, I'm not sure it's necessary for me to keep two sets of APIs, one with and one without an explicit transaction. But that would be a better done in a separate ticket.

Reply:
I just created a ticket [WT-12146](https://jira.mongodb.org/browse/WT-12146) that proposes to re-implement the non-transactional API as a wrapper around the transactional API.